libtelemetry/src/main/java/com/mapbox/services/android/telemetry/TelemetryService.java

```diff
@@ -167,4 +175,33 @@ TelemetryService obtainService() {
       return TelemetryService.this;
     }
   }
+
+  private boolean checkLocationPermission() {
+    if (PermissionsManager.areLocationPermissionsGranted(this)) {
+      return true;
+    } else {
+      permissionBackoff(this);
+      return false;
+    }
+  }
+
+  private void permissionBackoff(final Context context) {
+    final Handler handler = new Handler();
```

Comment:
This code may cause a memory leak.
👀 [How to Leak a Context: Handlers & Inner Classes](http://www.androiddesignpatterns.com/2013/01/inner-class-handler-memory-leak.html)